internal/collector/postgres_metrics.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 - 2025 Crunchy Data Solutions, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO fold this back into postgres.go once the collector package stabilizes.
+package collector
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	"github.com/crunchydata/postgres-operator/internal/feature"
+	"github.com/crunchydata/postgres-operator/internal/pgmonitor"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+// https://pkg.go.dev/embed
+//
+//go:embed "generated/postgres_5s_metrics.json"
+var defaultFiveSecondMetrics json.RawMessage
+
+//go:embed "generated/postgres_5m_metrics.json"
+var fiveMinuteMetrics json.RawMessage
+
+//go:embed "generated/pgbackrest_metrics.json"
+var pgBackRestMetrics json.RawMessage
+
+//go:embed "generated/gte_pg17_metrics.json"
+var gtePG17 json.RawMessage
+
+//go:embed "generated/lt_pg17_metrics.json"
+var ltPG17 json.RawMessage
+
+//go:embed "generated/gte_pg16_metrics.json"
+var gtePG16 json.RawMessage
+
+//go:embed "generated/lt_pg16_metrics.json"
+var ltPG16 json.RawMessage
+
+func EnablePostgresMetrics(ctx context.Context, inCluster *v1beta1.PostgresCluster, config *Config) {
+	if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+		var fiveSecondMetrics json.RawMessage
+		fiveSecondMetrics, _ = appendToJSONArray(defaultFiveSecondMetrics, pgBackRestMetrics)
```

Comment:
I need to think about error handling here.

Reply:
What are you questioning?

Reply:
I throw away the error here. I guess that's fine.